File: app/src/androidTest/java/tv/spacedentist/android/SDTestModule.java
package tv.spacedentist.android;

import com.google.android.gms.cast.framework.CastContext;

import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import tv.spacedentist.android.util.SDLogger;
import tv.spacedentist.android.util.SDLoggerAndroid;

public class SDTestModule extends SDModule {
    private static final SDLogger mLogger = new SDLoggerAndroid();
    @Mock private CastContext mCastApi;
    @Mock private SDNotificationManager mNotificationManager;

    public SDTestModule() {
        super(null);
        MockitoAnnotations.initMocks(this);
    }

    @Override
    CastContext provideCastContext() {
        return mCastApi;
    }

    @Override
    SDLogger provideLogger() {
        return mLogger;
    }

    @Override
    SDNotificationManager provideNotificationManager() {
        return mNotificationManager;
    }
}

File: app/src/androidTest/java/tv/spacedentist/android/SDTestModule.kt
package tv.spacedentist.android

import com.google.android.gms.cast.framework.CastContext
import org.mockito.Mock
import org.mockito.MockitoAnnotations
import tv.spacedentist.android.util.SDLogger
import tv.spacedentist.android.util.SDLoggerAndroid

class SDTestModule : SDModule(null) {
    @Mock private lateinit var mCastContext: CastContext
    @Mock private lateinit var mNotificationManager: SDNotificationManager

    init {
        MockitoAnnotations.initMocks(this)
    }

    override fun provideCastContext(): CastContext {
        return mCastContext
    }

    override fun provideLogger(): SDLogger {
        return mLogger
    }

    override fun provideNotificationManager(): SDNotificationManager {
        return mNotificationManager
    }

    companion object {
        private val mLogger = SDLoggerAndroid()
    }
}
